craft_parts/dirs.py

```diff
@@ -74,14 +75,25 @@ def __init__(
             )
         )
 
+    def _validate_requested_partition(
+        self, dir_name: str, partition: str | None = None
+    ) -> None:
+        """Ensure the requested partition is valid."""
+        if self._partitions:
+            if not partition:
+                raise PartitionUsageError(
+                    error_list=f"Partitions are enabled, you must specify which partition's {dir_name} you want.",
```

Comment:
```suggestion
                    error_list=[f"Partitions are enabled, you must specify which partition's {dir_name!r} you want."],
```

Thinking it makes more sense to say "...which partition's 'usr' you want" than "...which partitions usr you want."

But also, we probably want that as an iterable containing that one string rather than an iterable containing each character 🙂

Reply:
Oof, good catch!